From d1a39ac96c4d9d8c709a0b471521fff6ac685e3d Mon Sep 17 00:00:00 2001
From: Vinayakumar B <vinayakumarb@apache.org>
Date: Thu, 26 Feb 2015 14:36:09 +0530
Subject: [PATCH 1061/2748] HDFS-7832. Show 'Last Modified' in Namenode's
 'Browse Filesystem' (Contributed by Vinayakumar
 B)

(cherry picked from commit 166eecf687765f369cd5bf210cc6d7bf493121ea)
(cherry picked from commit f0f3d1937db2f43479b3daa6890dce1f86c1aa23)

Change-Id: Iecff863c0a68fc9a872e8531ab294540cdbe53a9
---
 .../src/main/webapps/hdfs/explorer.html            |    2 ++
 .../hadoop-hdfs/src/main/webapps/hdfs/explorer.js  |    9 ++++++++-
 2 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.html b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.html
index e1fdfa3..7b34044 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.html
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.html
@@ -112,6 +112,7 @@ <h4 class="modal-title" id="file-info-title">File information</h4>
             <th>Owner</th>
             <th>Group</th>
             <th>Size</th>
+            <th>Last Modified</th>
             <th>Replication</th>
             <th>Block Size</th>
             <th>Name</th>
@@ -124,6 +125,7 @@ <h4 class="modal-title" id="file-info-title">File information</h4>
             <td>{owner}</td>
             <td>{group}</td>
             <td>{length|fmt_bytes}</td>
+            <td>{#helper_date_tostring value="{modificationTime}"/}</td>
             <td>{replication}</td>
             <td>{blockSize|fmt_bytes}</td>
             <td><a style="cursor:pointer" inode-type="{type}" class="explorer-browse-links" inode-path="{pathSuffix}">{pathSuffix}</a></td>
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.js b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.js
index 87d47fa..0a53dcd 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.js
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.js
@@ -143,6 +143,12 @@
   }
 
   function browse_directory(dir) {
+    var HELPERS = {
+      'helper_date_tostring' : function (chunk, ctx, bodies, params) {
+        var value = dust.helpers.tap(params.value, chunk, ctx);
+        return chunk.write('' + new Date(Number(value)).toLocaleString());
+      }
+    };
     var url = '/webhdfs/v1' + dir + '?op=LISTSTATUS';
     $.get(url, function(data) {
       var d = get_response(data, "FileStatuses");
@@ -154,7 +160,8 @@
       current_directory = dir;
       $('#directory').val(dir);
       window.location.hash = dir;
-      dust.render('explorer', d, function(err, out) {
+      var base = dust.makeBase(HELPERS);
+      dust.render('explorer', base.push(d), function(err, out) {
         $('#panel').html(out);
 
         $('.explorer-browse-links').click(function() {
-- 
1.7.9.5

